Extend support for cluster/namespace scoping (#28)
keikoproj/active-monitor: README.md

- verify kube-dns by running DNS lookups on localhost
- verify KIAM agent by running aws sts get-caller-identity on all available nodes

> With the Cluster/Namespace level healtcheck can be run in any namespace provided namespace is already created.

Reviewer: ```suggestion
With the Cluster/Namespace level, healthchecks can be run in any namespace provided namespace is already created.
```

> With the Cluster/Namespace scoping healtcheck can be run in any namespace provided namespace is already created.

Reviewer: ```suggestion
With the Cluster/Namespace level healthcheck can be run in any namespace provided namespace is already created.
```

> The `level` in the workflow spec defines at which level the healtcheck runs it can be either namespace or cluster

Reviewer: ```suggestion
The `level` in the `HealthCheck` spec defines at which level it runs; it can be either `Namespace` or `Cluster`
```

> when `level` is set to namespace Active-Monitor will create a serviceaccount in the namespace as defined in the workflow spec, it will also create the role and rolebinding with namespace level permissions so that the healthchecks in a namespace can be performed.

Reviewer: ```suggestion
When `level` is set to `Namespace`, Active-Monitor will create a `ServiceAccount` in the namespace as defined in the workflow spec, it will also create the `Role` and `RoleBinding` with namespace level permissions so that the `HealthChecks` in a namespace can be performed.
```

## Usage and Examples
Create a new healthcheck:

> `kubectl create ns test`

Reviewer: how about if we provide this as a new example rather than overwriting the existing one

Author: sure will update with the changes

Events:                      <none>
```

## Example 2:

creating healthcheck in a new namespace

`kubectl create ns test`

`kubectl create -f https://raw.githubusercontent.com/keikoproj/active-monitor/master/examples/inlineHello_ns.yaml`

OR with local source code:

`kubectl create -f examples/inlineHello_ns.yaml`

Then, list all healthchecks:

`kubectl get healthcheck -n test` OR `kubectl get hc -n test`

```
NAME                 AGE
inline-hello-zz5vm   55s
```

View additional details/status of a healthcheck:

`kubectl describe healthcheck inline-hello-zz5vm -n test`

```
...
Status:
  Failed Count:              0
  Finished At:               2019-08-09T22:50:57Z
  Last Successful Workflow:  inline-hello-4mwxf
  Status:                    Succeeded
  Success Count:             13
Events:                      <none>
```

`argo list -n test`

```                                                                                                            
NAME                 STATUS      AGE   DURATION   PRIORITY
inline-hello-88rh2   Succeeded   29s   7s         0
inline-hello-xpsf5   Succeeded   1m    8s         0
inline-hello-z8llk   Succeeded   2m    7s         0

> inline-hello-j48cn   Succeeded   3m    8s         0

Reviewer: to save some scrolling in README, let's just cap the result set at 3, I think it conveys the same eitherway
```suggestion
```

> inline-hello-pcbhj   Succeeded   4m    7s         0

Reviewer: ```suggestion
```

> inline-hello-45cql   Succeeded   6m    7s         0

Reviewer: ```suggestion
```
